### PR TITLE
feature-1915/added back-end for working environment

### PR DESCRIPTION
**Pull request by MatthewBackes**
Added Back end for working environment section

NOTE BEFORE MERGING: Migration file is numbered 'V109', do not merge until ready

### CODE REVIEW

File: onboard-server/src/main/java/com/objectcomputing/checkins/services/onboard/onboardeeprofile/OnboardingProfileRepository.java [hunk 1]
```diff
@@ -9,50 +10,73 @@
...
+                        "PGP_SYM_DECRYPT(cast(mp.socialSecurityNumber as bytea),'${aes.key}') as socialSecurityNumber,"
+                        +
+                        "PGP_SYM_DECRYPT(cast(mp.birthDate as bytea),'${aes.key}') as birthDate," +
+                        "PGP_SYM_DECRYPT(cast(mp.currentAddress as bytea),'${aes.key}') as currentAddress," +
+                        "PGP_SYM_DECRYPT(cast(mp.previousAddress as bytea),'${aes.key}') as previousAddress," +
+                        "PGP_SYM_DECRYPT(cast(mp.phoneNumber as bytea),'${aes.key}') as phoneNumber," +
+                        "PGP_SYM_DECRYPT(cast(mp.secondPhoneNumber as bytea),'${aes.key}') as secondPhoneNumber," +
+                        "PGP_SYM_DECRYPT(cast(mp.personalEmail as bytea),'${aes.key}') as personalEmail " +
+                        "FROM \"onboard_profile\" mp " +
+                        "WHERE (onboard_profile_id = :onboardProfileId) ", nativeQuery = true)
```
Comment (S78901) [inline onboard-server/src/main/java/com/objectcomputing/checkins/services/onboard/onboardeeprofile/OnboardingProfileRepository.java:32]: Same issue with the id. Do a search for 'onboard_profile_id' and replace with id in these files.

File: onboard-server/src/main/java/com/objectcomputing/checkins/services/workingenvironment/WorkingEnvironment.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    public void setNewHireAccount(NewHireAccountEntity newHireAccount) {
+        this.newHireAccount = newHireAccount;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        WorkingEnvironment that = (WorkingEnvironment) o;
+        return Objects.equals(id, that.id) && Objects.equals(workLocation, that.workLocation) && Objects.equals(keyType, that.keyType) && Objects.equals(osType, that.osType) && Objects.equals(accessories, that.accessories) && Objects.equals(otherAccessories, that.otherAccessories) && Objects.equals(newHireAccount, that.newHireAccount);
```
Comment (S78901) [inline onboard-server/src/main/java/com/objectcomputing/checkins/services/workingenvironment/WorkingEnvironment.java:153]: NewHireAccount is left out of the equals and hashCode sections.

File: server/src/main/java/com/objectcomputing/checkins/services/workingenvironment/WorkingEnvironmentController.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import reactor.core.scheduler.Schedulers;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.net.URI;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+
+@Controller("/services/working-environment")
+@Secured(SecurityRule.IS_AUTHENTICATED)
```
Comment (S78901) [inline server/src/main/java/com/objectcomputing/checkins/services/workingenvironment/WorkingEnvironmentController.java]: Replace line 25 with:

`@Secured(RoleType.Constants.HR_ROLE)`

... and import:

`import com.objectcomputing.checkins.services.role.RoleType;`

File: server/src/main/java/com/objectcomputing/checkins/services/workingenvironment/WorkingEnvironment.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    public void setNewHireAccount(NewHireAccountEntity newHireAccount) {
+        this.newHireAccount = newHireAccount;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        WorkingEnvironment that = (WorkingEnvironment) o;
+        return Objects.equals(id, that.id) && Objects.equals(workLocation, that.workLocation) && Objects.equals(keyType, that.keyType) && Objects.equals(osType, that.osType) && Objects.equals(accessories, that.accessories) && Objects.equals(otherAccessories, that.otherAccessories) && Objects.equals(newHireAccount, that.newHireAccount);
```
Comment (S78901) [inline server/src/main/java/com/objectcomputing/checkins/services/workingenvironment/WorkingEnvironment.java:154]: We can leave NewHireAccount out of the hashCode and equals.

File: onboard-server/src/main/java/com/objectcomputing/checkins/services/onboard/onboardeeprofile/OnboardingProfileRepository.java [hunk 5]
```diff
@@ -9,50 +10,73 @@
...
+        Mono<OnboardingProfile> findById(@NotNull UUID onboardProfileId);
 
-    Optional<OnboardingProfile> findBySocialSecurityNumber(@NotNull String socialSecurityNumber);
+        Flux<OnboardingProfile> findAll();
 
-    Flux<OnboardingProfile> findAll();
+        @Query(value = "SELECT id, " +
```
Comment (S78901) [inline onboard-server/src/main/java/com/objectcomputing/checkins/services/onboard/onboardeeprofile/OnboardingProfileRepository.java:37]: Looks like we need to change this back. My bad.

File: onboard-server/src/main/java/com/objectcomputing/checkins/services/onboard/onboardeeprofile/OnboardingProfileRepository.java [hunk 6]
```diff
@@ -9,50 +10,73 @@
...
+                        +
+                        "AND  (:previousAddress IS NULL OR PGP_SYM_DECRYPT(cast(mp.previousAddress as bytea),'${aes.key}') = :previousAddress) "
+                        +
+                        "AND  (:phoneNumber IS NULL OR PGP_SYM_DECRYPT(cast(mp.phoneNumber as bytea),'${aes.key}') = :phoneNumber) "
+                        +
+                        "AND  (:secondPhoneNumber IS NULL OR PGP_SYM_DECRYPT(cast(mp.secondPhoneNumber as bytea),'${aes.key}') = :secondPhoneNumber) "
+                        +
+                        "AND  (:personalEmail IS NULL OR PGP_SYM_DECRYPT(cast(mp.personalEmail as bytea),'${aes.key}') = :personalEmail) ", nativeQuery = true)
+        Mono<OnboardingProfile> search(
+                        @Nullable String id,
```
Comment (S78901) [inline onboard-server/src/main/java/com/objectcomputing/checkins/services/onboard/onboardeeprofile/OnboardingProfileRepository.java:70]: & here

File: server/src/main/java/com/objectcomputing/checkins/services/onboard/onboardeeprofile/OnboardingProfileRepository.java [hunk 7]
```diff
@@ -9,50 +10,73 @@
...
+                        "PGP_SYM_DECRYPT(cast(mp.secondPhoneNumber as bytea),'${aes.key}') as secondPhoneNumber," +
+                        "PGP_SYM_DECRYPT(cast(mp.personalEmail as bytea),'${aes.key}') as personalEmail " +
+                        "FROM \"onboard_profile\" mp " +
+                        "WHERE (onboard_profile_id = :onboardProfileId) ", nativeQuery = true)
+        Mono<OnboardingProfile> findById(@NotNull UUID onboardProfileId);
+
+        Flux<OnboardingProfile> findAll();
 
-    Flux<OnboardingProfile> findAll();
+        @Query(value = "SELECT id, " +
```
Comment (S78901) [inline server/src/main/java/com/objectcomputing/checkins/services/onboard/onboardeeprofile/OnboardingProfileRepository.java:37]: & here

File: server/src/main/java/com/objectcomputing/checkins/services/onboard/onboardeeprofile/OnboardingProfileRepository.java [hunk 8]
```diff
@@ -9,50 +10,73 @@
...
+                        +
+                        "AND  (:previousAddress IS NULL OR PGP_SYM_DECRYPT(cast(mp.previousAddress as bytea),'${aes.key}') = :previousAddress) "
+                        +
+                        "AND  (:phoneNumber IS NULL OR PGP_SYM_DECRYPT(cast(mp.phoneNumber as bytea),'${aes.key}') = :phoneNumber) "
+                        +
+                        "AND  (:secondPhoneNumber IS NULL OR PGP_SYM_DECRYPT(cast(mp.secondPhoneNumber as bytea),'${aes.key}') = :secondPhoneNumber) "
+                        +
+                        "AND  (:personalEmail IS NULL OR PGP_SYM_DECRYPT(cast(mp.personalEmail as bytea),'${aes.key}') = :personalEmail) ", nativeQuery = true)
+        Mono<OnboardingProfile> search(
+                        @Nullable String id,
```
Comment (S78901) [inline server/src/main/java/com/objectcomputing/checkins/services/onboard/onboardeeprofile/OnboardingProfileRepository.java:70]: & here

**Review by S78901 — DISMISSED** [review]

**Review by S78901 — APPROVED** [review]

**Comment by mkimberlin** [comment]
@S78901 @MatthewBackes @1mbalducci I'm bringing this one down into develop. It should set us up to get the others corrected.

**Pull request merged**